bp-groupblog.php

```diff
@@ -1718,4 +1718,42 @@ function bp_groupblog_delete_meta( $blog_id, $drop = false ) {
 }
 add_action('delete_blog', 'bp_groupblog_delete_meta', 10, 1);
 
-?>
+/**
+ * Use the group avatar on the Site Directory page for groupblogs.
+ *
+ * If a site in the site loop is a groupblog, use the group logo only if
+ * the site doesn't already have a customized site icon.
+ *
+ * @since 1.9.2
+ *
+ * @param  string $retval  Current site avatar
+ * @param  int    $blog_id Site ID in loop
+ * @param  array  $r       Avatar arguments
+ * @return string
+ */
+function bp_groupblog_use_group_avatar_in_site_loop( $retval, $blog_id, $r ) {
+	// Not a groupblog? Bail.
+	$group_id = get_groupblog_group_id( $blog_id );
+	if ( empty( $group_id ) ) {
+		return $retval;
+	}
+
+	// Already using a site icon, so bail.
```

Comment:
Is this really the only way to check? Feels a bit fragile to me. Might be an interesting enhancement in BuddyPress - something like a `data-avatar-type="site-admin-avatar"` or `"site-icon"` attribute, which could more reliably be sniffed here. Not a dealbreaker, just something that jumped out at me :)

Reply:
Ahh, yes, good call! We can also check blogmeta.

Check out the latest commit in the PR - 01ad8d7.

Reply:
Yeah, this feel safer to me. Thanks!